mmda/types/annotation.py

```diff
@@ -95,27 +116,26 @@ def to_json(self) -> Dict:
 class Indexer:
     """Stores an index for a particular collection of Annotations.
     Indexes in this library focus on *INTERSECT* relations."""
-    raise NotImplementedError
 
+    @abstractmethod
+    def find(self, query: Annotation) -> List[Annotation]:
+        """Returns all matching Annotations given a suitable query"""
 
-@dataclass
-class DocSpanGroupIndexer(Indexer):
 
-    num_pages: int
+@dataclass
+class SpanGroupIndexer(Indexer):
 
-    # TODO[kylel] -- explain why?
     def __post_init__(self):
-        self._index: List[IntervalTree] = [IntervalTree() for _ in range(self.num_pages)]
+        self._index: IntervalTree = IntervalTree()
```

Comment:
How about having `_index` in the dataclass init?  
```python
class SpanGroupIndexer(Indexer):
    _index: IntervalTree = IntervalTree()

```

Reply:
done